engine/src/main/scala/cromwell/engine/workflow/lifecycle/ShadowMaterializeWorkflowDescriptorActor.scala

```diff
@@ -0,0 +1,300 @@
+    } yield EngineWorkflowDescriptor(backendDescriptor, backendAssignments, failureMode)
+  }
+
+  private def validateBackendAssignments(calls: Seq[Call], workflowOptions: WorkflowOptions): ErrorOr[Map[Call, String]] = {
+    val assignmentAttempt = Try {
+      val backendFromOptions = workflowOptions.get(RuntimeBackendKey)
+      val callAssignmentMap = backendFromOptions match {
+        case Success(backendName) =>
+          // Found an entry for backendKey in the workflow options.
+          // Use this backend for all the calls in the workflow
+          // If this backend is not in the list of plugged in backends, return the default one for the whole workflow
+          calls.map(_ -> validateBackend(backendName)).toMap[Call, String]
+        case Failure(reason) =>
+          log.warning("Backend was not defined in the workflow options, trying for runtime-attributes based per call backend allocation.", reason)
+          (calls zip (calls map mapCallToBackend)).toMap[Call, String]
+      }
+      log.debug(s"$tag: Call-to-Backend assignments: $callAssignmentMap")
+      callAssignmentMap
+    }
+
+    assignmentAttempt match {
+      case Success(x) => x.successNel
+      case Failure(t) => t.getMessage.failureNel
+    }
+  }
+
+  /**
+    * Map a call to a backend name depending on the runtime attribute key
+    */
+  private def mapCallToBackend(call: Call): String = {
+
+    val runtimeAttributesMap = call.task.runtimeAttributes.attrs
+    val backend = runtimeAttributesMap.get(RuntimeBackendKey) match {
+      case Some(backendNameAsExp) =>
+        // Don't allow *any* lookup or functions. The runtime must be static!
+        val backendNameAsString: String = backendNameAsExp.evaluate(NoLookup, NoFunctions) match {
+          case Success(runtimeString: WdlString) => runtimeString.valueString
+          case Success(x: WdlValue) =>
+            throw new Exception(s"Non-string values are not currently supported for backends! Cannot assign backend '${x.valueString}' to backend to Call: ${call.fullyQualifiedName}")
+          case Failure(error) =>
+            throw new Exception(s"Dynamic backends are not currently supported! Cannot assign backend '${backendNameAsExp.valueString}' for Call: ${call.fullyQualifiedName}", error)
+        }
+        backendNameAsString
+      case None =>
+        log.info("Using default backend ")
+        DefaultBackend
+    }
+    log.info(s"Mapped Call [${call.fullyQualifiedName}] to Backend [$backend] ")
+    validateBackend(backend)
+  }
+
+  /**
+    * Checks if a given backend name is plugged-in in to the Cromwell sub-system and returns if true.
+    * Returns the default backend if `backendName` is not found in the list
+     */
+  private def validateBackend(backendName: String): String = {
+    if (PluggedInBackends.exists(_.equalsIgnoreCase(backendName)))
+      backendName
+    else {
```

Comment:
don't mix the non-brace on the if with braces on the else. I prefer with braces but the scala style guide disagrees with me.